src/analysis/convex/basic.lean

```diff
@@ -351,6 +352,103 @@ convex_halfspace_gt (is_linear_map.mk complex.add_im complex.smul_im) _
 lemma convex_halfspace_im_lge (r : ℝ) : convex {c : ℂ | r ≤ c.im} :=
 convex_halfspace_ge (is_linear_map.mk complex.add_im complex.smul_im) _
 
+/- Convex combinations in intervals -/
+
+lemma convex.combo_self (a : α) {x y : α} (h : x + y = 1) : a = x * a + y * a :=
+  calc
+      a   = 1 * a           : by rw[one_mul]
+      ... = (x + y) * a     : by rw[h]
+      ... = x * a + y * a   : by rw[add_mul]
```

Comment:
Here and below, please put a space between `rw` (or `nlinarith`, `linarith`, etc.) and the list of arguments:
```suggestion
      a   = 1 * a           : by rw [one_mul]
      ... = (x + y) * a     : by rw [h]
      ... = x * a + y * a   : by rw [add_mul]
```